osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs

```diff
@@ -60,7 +62,12 @@ protected override void Update()
         {
             base.Update();
 
-            if (!IsBreakTime.Value)
+            if (IsBreakTime.Value)
+            {
+                breakendcounter = 0;
+            }
+
+            if (!IsBreakTime.Value && breakendcounter > 0)
```

Comment:
```csharp
if (IsBreakTime.Value || !judgedAfterBreak)
{
    judgedAfterBreak = false;
    return;
}

// draining logic
...
```

Reply:
also provide a comment as to why draining will only be applied when a hit object is judged post-break.

Reply:
!judgedAfterBreak and judgedAfterBreak = false is the same thing or am I wrong here?

Reply:
! is faster than ==

Reply:
I think I would only add `return;` because I can't seem to grasp the` ...|| !judgedAfterBreak and judgedAfterBreak = false` part.

Reply:
You can get to the assignment via the other variable in the or operator

Reply:
He means that there's a simple oversight in the suggestion, making it essentially

```csharp
if (!judgedAfterBreak)
    judgedAfterBreak = false;
```

Should've been without the `!` here.

Reply:
@kornkaobat I've suggested it for better code readability:

current code: 
1. if on break time, clear `judgedAfterBreak`. 
2. if not on break time and is `judgedAfterBreak` then do draining logic.

suggested code: 
1. if on break time or not `judgedAfterBreak` then clear `judgedAfterBreak` and don't do draining logic (aka `return;`). 
2. do draining logic

also clearing `judgedAfterBreak` inside the check is intended because we're featuring `IsBreakTime` on the check and don't really like breaking it to 2 separate `if`s.

Reply:
Fixed via bdach's suggestion e09786e.

Reply:
Why aren't you using the "resolve conversation" button?

Reply:
Already done that.